src/main/java/io/kestra/plugin/notion/ReadPage.java

```
@@ -0,0 +1,119 @@
+    public Output run(RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+
+        try {
+            // Validate and render page ID
+            String renderedPageId = validateAndRenderPageId(runContext);
+            
+            logger.info("Reading Notion page with ID: {}", renderedPageId);
+
+            // Get page metadata
+            String pageUrl = buildPageURL(renderedPageId);
+            HttpRequest.HttpRequestBuilder pageRequestBuilder = buildGetRequest(runContext, pageUrl);
+            
+            NotionResponse pageResponse = makeCall(runContext, pageRequestBuilder, NotionResponse.class);
+            
+            // Get page content (blocks/children)
+            String childrenUrl = buildPageChildrenURL(renderedPageId);
+            HttpRequest.HttpRequestBuilder childrenRequestBuilder = buildGetRequest(runContext, childrenUrl);
+            
+            NotionResponse childrenResponse = makeCall(runContext, childrenRequestBuilder, NotionResponse.class);
+            
+            // Convert blocks to markdown
+            String markdownContent = MarkdownConverter.blocksToMarkdown(mapper.valueToTree(childrenResponse.getChildren()));
+            
+            // Store detailed information
+            URI fileURI = store(runContext, List.of(Map.of(
+                "pageResponse", pageResponse,
+                "childrenResponse", childrenResponse,
+                "markdownContent", markdownContent,
+                "pageId", renderedPageId
+            )));
+            
+            return ((Output.OutputBuilder) buildCommonOutput(pageResponse))
+                .content(markdownContent)
+                .uri(fileURI)
+                .message("Page read successfully")
+                .build();
+                
+        } catch (Exception e) {
+            logger.error("Error reading Notion page: {}", e.getMessage());
+            return buildErrorOutput("Failed to read page: " + e.getMessage());
```

Maybe throw an exception rather than returning an error output; otherwise this will make the task `SUCCESS` with errors.

---

fixed by https://github.com/kestra-io/plugin-notion/pull/2/commits/01c9963a8ad9e3087a05cd238fae2e287a80414b